File: providers/custom.js
module.exports = {
  
  name: 'custom',
  aliases: ['ollama', 'vllm', 'local'],
  displayName: 'Custom (OpenAI-compatible)',
  description: 'Generic OpenAI-compatible endpoint — Ollama, vLLM, llama.cpp, etc.',
  baseUrl: 'http://localhost:11434/v1',
  apiMode: 'openai_chat',
  envVars: ['CUSTOM_API_KEY', 'CUSTOM_BASE_URL'],
  authType: 'api_key',
  defaultModel: 'llama3',
  defaultMaxTokens: 4096,
  supportsHealthCheck: false,
  modelProfiles: {
    '*': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 32768, maxOutputTokens: 4096 },
  },

};

File: backend-py/app/providers/custom.py
"""
Provider config for Custom (OpenAI-compatible).
"""
from __future__ import annotations
INFO: dict[str, object] = {'name': 'Custom (OpenAI-compatible)', 'aliases': ['ollama', 'vllm', 'local'], 'display_name': 'Custom (OpenAI-compatible)', 'description': 'Generic OpenAI-compatible endpoint — Ollama, vLLM, llama.cpp, etc.', 'base_url': 'http://localhost:11434/v1', 'api_mode': 'openai_chat', 'env_vars': ['CUSTOM_API_KEY', 'CUSTOM_BASE_URL'], 'auth_type': 'api_key', 'default_model': 'llama3', 'default_max_tokens': 4096, 'model_profiles': {'*': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 32768, 'maxOutputTokens': 4096}}}

def resolveBaseUrl() -> str:
    return 'http://localhost:11434/v1'

def resolveApiKey(envKey: str | None=None) -> str | None:
    return envKey